### PR TITLE
initial auth_ip_geo analysis sandbox with cidr module

### PR DESCRIPTION
**Pull request by ameihm0912**

### CODE REVIEW

File: moz_security/tests/integration/analysis_auth_ip_geo/run/input/generate_auth.cfg [hunk 1]
```diff
@@ -0,0 +1,21 @@
+filename = "generate_auth.lua"
+
+maxminddb_heka = {
+    databases = {
+        ["GeoIP2-City-Test.mmdb"] = {
```
Comment (trink) [inline moz_security/tests/integration/analysis_auth_ip_geo/run/input/generate_auth.cfg]: We are starting to get multiple copies of this DB in the repo, should we just start referencing the one here https://github.com/mozilla-services/lua_sandbox_extensions/blob/master/maxminddb/tests/GeoIP2-City-Test.mmdb

File: moz_security/modules/iputils.lua [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+-- Precompute inverted binary subnet masks
+local bin_inverted_masks = {}
+for i=0,32 do
+    local i = tostring(i)
+    bin_inverted_masks[i] = bit.bxor(bin_masks[i], bin_masks["32"])
+end
+
+
+local function split_octets(input)
```
Comment (trink) [inline moz_security/modules/iputils.lua]: Generally I would use a grammar here (add it to ip_address grammar)
```
local octet =  d8 / tonumber
v4_octet = l.Ct(octet * "." *  octet * "." * octet * "." * octet)
```

File: moz_security/modules/iputils.lua [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+local string    = require("string")
+
+local M = {}
+setfenv(1, M)
+
+
+-- Precompute binary subnet masks
+local bin_masks = {}
+for i=0,32 do
+    bin_masks[tostring(i)] = bit.lshift((2^i)-1, 32-i)
```
[the remark this answers is not in the document]
Comment (ameihm0912) [reply moz_security/modules/iputils.lua]: In parse_cidr the mask is accessed through splitting the string (e.g., `"0.0.0.0/0" -> { "0.0.0.0", "0" } -> bin_masks["0"]`). This comes from the code this was based on, we can probably change it if required though.

File: moz_security/modules/iputils.lua [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    return bin
+end
+
+
+local function ip2bin(ip)
+    if type(ip) ~= "string" then
+        return nil
+    end
+
+    local octets = split_octets(ip)
```
Comment (trink) [inline moz_security/modules/iputils.lua]: in-line it?

File: moz_security/modules/iputils.lua (hunk 3 above)
[the remark this answers is not in the document]
Comment (trink) [reply moz_security/modules/iputils.lua]: We have access to the integer value everywhere (even after it is parsed out of the string it is tonumber'ed below) so we get rid of a bunch of conversions and tmp variables.  Your call.

File: moz_security/modules/iputils.lua [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    if not octets or #octets ~= 4 then
```
Comment (trink) [inline moz_security/modules/iputils.lua]: Don't need the length check the grammar verified it (the grammar could actually do the complete conversion and this function could go away (optional)).

**Pull request merged**